### PR TITLE
docs: ADRish describing the initial model and API

### PR DESCRIPTION
**Pull request by ashultz0**
**JIRA:** https://2u-internal.atlassian.net/browse/MST-1499

**Description:** This adds an ADR which roughly describes the initial model and API. Condensed from https://2u-internal.atlassian.net/wiki/spaces/PT/pages/15438388/Program+Intent+Technical+Plan

**Author concerns:** This doesn't at all follow the ADR format because that didn't seem helpful to me but you may disagree.

Also the API is specified in very handwavy terms, I think that's OK at this stage.

### CODE REVIEW

File: docs/decisions/0002-pie-structure.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+Intents are never modified. If new information gives us a new belief about the learner's intentions, we write a new intent for that learner and program, leaving the old one in place. Intents have two timestamps to assist with ordering, the standard **created** timestamp and an **effective** timestamp which is the time this intent became true to the system. If a weekly batch job on Saturday were to notice that the user had signed up for a program on Tuesday, created would be Saturday, and effective would be Tuesday.
+
+Model
++++++
+
++--------------------+----------------------------------------+
+|Data                |Note                                    |
++====================+========================================+
+|Program ID          |program UUID is the generic identifier  |
```
Comment (MichaelRoytman) [inline docs/decisions/0002-pie-structure.rst:22]: This has a relationship to the program UUID in discovery. Is that worth calling out here or too much an implementation detail?

In any case, what do you mean by "generic" here?

File: docs/decisions/0002-pie-structure.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+Writes
+++++++
+
+
+MVP
+~~~
+
+We need a simple write event call, something like:
+
+POST /programintent (user ID, program ID, type, category, optional effective time if not NOW)
```
Comment (MichaelRoytman) [inline docs/decisions/0002-pie-structure.rst:52]: `type` = `reason`?

File: docs/decisions/0002-pie-structure.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+To bootstrap our intents we also need a management command to load intents generated by reports:
+
+LoadAllTheseProgramIntents (intent file)
+
+
+Later Work
+~~~~~~~~~~
+
+As a replacement for the management command we will connect a pipeline from the data warehouse to the PIE service. This will remove the periodic manual update we'll be using during the MVP.
```
Comment (MichaelRoytman) [inline docs/decisions/0002-pie-structure.rst:64]: What is the "periodic manual update" here? Is the assumption we'll periodically run the `loadAllTheseProgramIntents` management command up until this point?

File: docs/decisions/0002-pie-structure.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+Even later when event bus is ready, the PIE service can listen to enrollment or purchase events to immediately mark some intents. This would let us give quick feedback that we can’t do via batch job even if the user is not directly telling us their intent. We could do this work by adding intent code all over enrollments and purchase but that would make too much of the code PIES aware and tangle everything together.
+
+
+Reads
++++++
+
+MVP
+~~~
+
+Our first projected use for program intent is UI customization. For that we want a read call which prefers definitive intents over maybes.
```
Comment (MichaelRoytman) [inline docs/decisions/0002-pie-structure.rst:75]:
```suggestion
Our first projected use for program intent is UI customization. For that we want a read call which prefers certain intents over maybes.
```

Just to echo back to the data model.

File: docs/decisions/0002-pie-structure.rst (hunk 3 above)
Comment (ashultz0) [reply docs/decisions/0002-pie-structure.rst:64]: that's exactly it - for the MVP we'll have to just carry these in a bucket

**Review by MichaelRoytman — APPROVED** [review]
Thanks for addressing my feedback! 🚀

**Pull request merged**